Code/Framework/AzFramework/AzFramework/Spawnable/Script/SpawnableScriptAssetRef.cpp

```diff
@@ -80,14 +82,19 @@ namespace AzFramework::Scripts
     {
         if (this != &rhs)
         {
-            m_asset = AZStd::move(rhs.m_asset);
+            SetAsset(AZStd::move(rhs.m_asset));
         }
         return *this;
     }
 
     void SpawnableScriptAssetRef::SetAsset(const AZ::Data::Asset<Spawnable>& asset)
     {
+        AZ::Data::AssetBus::Handler::BusDisconnect();
         m_asset = asset;
+        if (m_asset.GetId().IsValid())
+        {
+            AZ::Data::AssetBus::Handler::BusConnect(m_asset.GetId());
+        }
```

Comment:
I think you still need to queue the asset and handle OnAssetReady. I'm not sure that you can guarantee the it will always be ready before hand. If that is the assumption then you need to assert that asset.IsReady().